File: src/codility/Iterations/BinaryGap.java
package codility.Iterations;

public class BinaryGap {

    public static void main(String[] args) {
        new BinaryGap().solution(1041);
    }

    public int solution(int N) {
        int ans = 0;
        int gap = 0;
        boolean flag = false;
        while (N > 0) {
            int mod = N %2;
            if (flag) {
                if (mod == 1) {
                    ans = Math.max(ans, gap);
                    gap = 0;
                } else {
                    gap++;
                }
            } else {
                if (mod == 1) {
                    flag = true;
                }
            }
            N=  N / 2;
        }
        return ans;
    }
}

File: src/codility/kotlin/BinaryGap.kt
package codility.kotlin

fun main() {
    print(solution(1041 ))
}

fun solution(N: Int): Int {
    var start = false

    var current = N
    var length = 0
    var maxLength = 0

    while (current > 0) {
        if (current % 2 == 1) {
            if (!start) {
                start = true
            } else {
                maxLength = Math.max(maxLength, length)
                length = 0
            }

        } else {
            if (start) {
                length++
            }
        }

        current = current /2

    }
    return maxLength

}
